File: MML-Card-BE/Models/Card/Useable/Equipment.cs
using MML_Card_BE.Models.Card.Useable.Interfaces;
using MML_Card_BE.Models.Enum;
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson;

namespace MML_Card_BE.Models.Card.Useable
{
    public class Equipment : IEquipment
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string? _id { get; set; }
        
        [BsonElement("name")]
        public string Name { get; set; }
        
        [BsonElement("description")]
        public string? Description { get; set; }
        
        [BsonElement("cardType")]
        private CardTypeEnum CardType { get; set; }
        public string type
        {
            get => CardType.ToString();
        }

        [BsonElement("imageId")]
        [BsonRepresentation(BsonType.String)]
        public string? ImageId { get; set; }

        [BsonElement("slot")]
        [BsonRepresentation(BsonType.Int32)]
        public EquipmentSlot Slot { get; set; }

        [BsonElement("bonus")]
        [BsonRepresentation(BsonType.Int32)]
        public int? Bonus { get; set; }

        [BsonElement("cardNumber")]
        [BsonRepresentation(BsonType.Int32)]
        public int CardNumber { get; set; }

        [BsonElement("CardSet")]
        [BsonRepresentation(BsonType.String)]
        public SetEnum Set { get; set; }
    }
}

File: card-web-client/src/models/Useable/Equipment.ts
import { ICardBase } from '../CardBase';
import CardTypeEnum from '../Enum/CardTypeEnum';
import EquipmentSlotEnum from '../Enum/EquipmentSlotEnum';

interface IEquipment extends ICardBase {
  bonus: number;
}

class Equipment implements IEquipment {
  _id: number;
  name: string;
  description: string;
  private cardType: CardTypeEnum;
  public get type(): string {
    return CardTypeEnum[this.cardType];
  }
  public set type(value: string) {
    this.cardType = CardTypeEnum[value as keyof typeof CardTypeEnum];
  }
  imageId: string;

  private slot: EquipmentSlotEnum;
  public get equipmentSlot(): string {
    return EquipmentSlotEnum[this.slot];
  }
  private set equipmentSlot(value: string) {
    this.slot = EquipmentSlotEnum[value as keyof typeof EquipmentSlotEnum];
  }
  bonus: number;

  constructor(
    _id: number,
    name: string,
    description: string,
    cardType: CardTypeEnum,
    imageId: string,
    slot: EquipmentSlotEnum,
    bonus: number
  ) {
    this._id = _id;
    this.name = name;
    this.description = description;
    this.cardType = cardType;
    this.imageId = imageId;
    this.slot = slot;
    this.bonus = bonus;
  }
}

export default Equipment;
